src/types.ts

```diff
@@ -86,15 +86,66 @@ export type Blockchain =
       ens?: NamingServiceSource;
       zns?: NamingServiceSource;
       cns?: NamingServiceSource;
-      web3Provider?: Web3Provider;
+      web3Provider?: Provider;
+      provider?: Provider;
     };
 
 export type API = {
   url: string;
 };
+type ProviderParams = readonly [
+  (
+    | {
+        [record: string]: string;
+      }
+    | string
+  ),
+  string,
+];
+
+export interface RequestArguments {
+  method: string;
+  params?: unknown[] | object;
+}
+
+export interface Provider {
+  call: (method: string, params: ProviderParams) => Promise<any>;
```

Comment:
I'm thinking now that maybe it's worth to just introduce EIP-1193 interface to our library and wrap all the supported adapters into it, instead of introducing "YetAnotherProviderInterface"?

Reply:
It sounds like a good idea! When EIP-1193 will go out we will already be supporting that out of the box